src/enforcer.ts

```diff
@@ -553,18 +553,18 @@ export class Enforcer {
       }
 
       return ruleRemoved;
-    } else if (sec instanceof Array) {
-      return await this.removeNamedPolicy('p', sec);
     } else {
-      return false;
+      return await this.removeNamedPolicy('p', ...params);
     }
   }
 
   /**
    * removeFilteredPolicy removes rules based on field filters from the current policy.
    */
-  public async removeFilteredPolicy(sec: string | number, key: string | string[], fieldIndex?: number, fieldValues?: string[]): Promise<boolean> {
-    if (typeof sec === 'string' && typeof key === 'string' && fieldIndex && fieldValues instanceof Array) {
+  // public async removeFilteredPolicy(sec: string | number, key: string | string[], fieldIndex?: number, fieldValues?: string[]): Promise<boolean> {
+  public async removeFilteredPolicy(...params: any[]): Promise<boolean> {
```

Comment:
Why is this changed? The parameters become unclear now.

Reply:
Like go APIs: 
In Public AddPolicy method, overload private addPolicy method.